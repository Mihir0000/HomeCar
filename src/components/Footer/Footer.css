
.footer_section {
    width: 100%;
    float: left;
    background-color: #252525;
    height: auto;
    padding-bottom: 50px;
    box-shadow: 0px 24px 19px -1px #222223;
    position: relative;
}
.layout_padding {
    padding-top: 90px;
    padding-bottom: 0px;
}

.footer_logo {
    width: 80%;
    float: left;
    padding: 0px 0px 30px 0px;
}

.adderss_text {
    width: 100%;
    float: left;
    font-size: 24px;
    color: #ffffff;
    padding-bottom: 30px;
}

.map_icon {
    width: 100%;
    float: left;
    font-size: 16px;
    color: #d6d6d6;
    padding-bottom: 10px;
}

.paddlin_left_0 {
    padding-left: 15px;
}

.hiphop_text_1 {
    width: 90%;
    color: #d6d6d6;
    font-size: 16px;
    margin-left: 0px;
}
.Useful_text {
    width: 50%;
    color: #d6d6d6;
    font-size: 16px;
    margin-left: 0px;
}

.Enter_text {
    width: 100%;
    float: left;
    font-size: 18px;
    color: #ffffff;
    background-color: transparent;
    padding: 7px 10px;
    margin-top: 10px;
    border: 1px solid #cfcfd0;
}

input.Enter_text::placeholder {
    color: #fff;
}

.subscribe_bt {
    width: 170px;
    float: left;
    margin-top: 20px;
}

.subscribe_bt a {
    width: 100%;
    float: left;
    font-size: 18px;
    color: #ffffff;
    background-color: #19e9d9;
    text-align: center;
    padding: 7px 0px;
    text-transform: uppercase;
}
.social_icon {
    width: 100%;
    float: left;
}

.social_icon ul {
    margin: 0px;
    padding: 0px;
}

.social_icon li {
    float: left;
    padding-right: 10px;
    padding-top: 20px;
}

.copyright_section {
    width: 100%;
    float: left;
    background-color: #252525;
    height: auto;
}

.copyright_text {
    width: 100%;
    float: left;
    font-size: 16px;
    margin-left: 0px;
    color: #fff;
    text-align: center;
}

.copyright_text a {
    color: #ffffff;
}

.copyright_text a:hover {
    color: #19e9d9;
}