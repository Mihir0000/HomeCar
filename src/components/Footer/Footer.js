import React from 'react';
import { Card, Col, Container, Form, Image, Nav, Row } from 'react-bootstrap';
import './Footer.css';

const Footer = ({ email, phone, address }) => {
    return (
        <>
            <Container fluid className="footer_section layout_padding">
                <Container>
                    <Row>
                        <Col lg={3} sm={6}>
                            <Card
                                className="footer_logo bg-transparent"
                                style={{ border: 'none' }}
                            >
                                <Card.Link href="index.html">
                                    <Image
                                        src="static/img/footer-logo.png"
                                        alt="footer"
                                    />
                                </Card.Link>
                            </Card>
                            <Card.Text className="adderss_text fs-2">
                                Contact Us
                            </Card.Text>
                            <Card.Title className="map_icon d-flex">
                                <Image
                                    src="static/img/map-icon.png"
                                    alt="map"
                                />
                                <Card.Subtitle className="paddlin_left_0">
                                    {address}
                                </Card.Subtitle>
                            </Card.Title>
                            <Card.Title className="map_icon d-flex">
                                <Image
                                    src="static/img/call-icon.png"
                                    alt="call"
                                />
                                <Card.Subtitle className="paddlin_left_0">
                                    {phone}
                                </Card.Subtitle>
                            </Card.Title>
                            <Card.Title className="map_icon d-flex">
                                <Image
                                    src="static/img/mail-icon.png"
                                    alt="mail"
                                />
                                <Card.Subtitle className="paddlin_left_0">
                                    {email}
                                </Card.Subtitle>
                            </Card.Title>
                        </Col>
                        <Col lg={3} sm={6}>
                            <Card.Text className="adderss_text fs-2">
                                Doctors
                            </Card.Text>
                            <Card.Text className="hiphop_text_1">
                                There are many variations of passages of Lorem
                                Ipsum available, but the majority have suffered
                                alteration in some form, by injected humour,
                            </Card.Text>
                        </Col>
                        <Col lg={3} sm={6}>
                            <Card.Text className="adderss_text fs-2">
                                Useful Links
                            </Card.Text>
                            <Card.Text className="Useful_text">
                                There are many variations of passages of Lorem
                                Ipsum available, but the majority have suffered
                                ,
                            </Card.Text>
                        </Col>
                        <Col lg={3} sm={6}>
                            <Card.Text className="adderss_text fs-2">
                                Newsletter
                            </Card.Text>
                            <Form.Control
                                className="Enter_text bg-transparent"
                                placeholder="Enter Your Email"
                                aria-label="Enter Your Email"
                                aria-describedby="basic-addon2"
                            />

                            <Card.Text className="subscribe_bt">
                                <Card.Link href="/#">Subscribe</Card.Link>
                            </Card.Text>
                            <Card
                                className="social_icon py-3 bg-transparent"
                                style={{ border: 'none' }}
                            >
                                <Nav>
                                    <Nav.Item>
                                        <Nav.Link href="/#" className="p-1 m-0">
                                            <Image
                                                src="static/img/fb-icon.png"
                                                alt="fb"
                                            />
                                        </Nav.Link>
                                    </Nav.Item>
                                    <Nav.Item>
                                        <Nav.Link href="/#" className="p-1 m-0">
                                            <Image
                                                src="static/img/twitter-icon.png"
                                                alt="twitter"
                                            />
                                        </Nav.Link>
                                    </Nav.Item>
                                    <Nav.Item>
                                        <Nav.Link href="/#" className="p-1 m-0">
                                            <Image
                                                src="static/img/linkedin-icon.png"
                                                alt="linkedin"
                                            />
                                        </Nav.Link>
                                    </Nav.Item>
                                    <Nav.Item>
                                        <Nav.Link href="/#" className="p-1 m-0">
                                            <Image
                                                src="static/img/instagram-icon.png"
                                                alt="instagram"
                                            />
                                        </Nav.Link>
                                    </Nav.Item>
                                </Nav>
                            </Card>
                        </Col>
                    </Row>
                </Container>
            </Container>
            <Container fluid className="copyright_section">
                <Container>
                    <Card className="copyright_text align-self-center bg-transparent flex-row justify-content-center fs-5">
                        2022 All Rights Reserved. Design by
                        <Card.Link href="https://html.design">
                            Free html Templates
                        </Card.Link>
                    </Card>
                </Container>
            </Container>
        </>
    );
};

export default Footer;
