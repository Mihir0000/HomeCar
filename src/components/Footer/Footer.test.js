import React from 'react';
import { render, screen } from '@testing-library/react';
import Footer from './Footer';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

test('footer renders', () => {
    render(<Footer />);
    const linkElement = screen.getByText(/Useful Links/i);
    expect(linkElement).toBeInTheDocument();
});

test('snapshot testing --Footer', () => {
    const component = renderer.create(<Footer />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Subscribe email enter', () => {
    render(<Footer />);
    const inputEmail = screen.getByPlaceholderText(/Enter Your Email/i);
    expect(inputEmail).toBeInTheDocument();
});

it('Should be components with footer', ()=>{
   const wrapper = shallow(<Footer email="abc@gmail.com" phone='1234567890' address='madhyamgram'/>);
   expect(wrapper).toMatchSnapshot();
})