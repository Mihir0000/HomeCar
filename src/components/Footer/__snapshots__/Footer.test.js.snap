// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be components with footer 1`] = `
<Fragment>
  <Container
    className="footer_section layout_padding"
    fluid={true}
  >
    <Container
      fluid={false}
    >
      <Row>
        <Col
          lg={3}
          sm={6}
        >
          <Card
            body={false}
            className="footer_logo bg-transparent"
            style={
              Object {
                "border": "none",
              }
            }
          >
            <CardLink
              href="index.html"
            >
              <Image
                alt="footer"
                fluid={false}
                rounded={false}
                roundedCircle={false}
                src="static/img/footer-logo.png"
                thumbnail={false}
              />
            </CardLink>
          </Card>
          <CardText
            className="adderss_text fs-2"
          >
            Contact Us
          </CardText>
          <CardTitle
            className="map_icon d-flex"
          >
            <Image
              alt="map"
              fluid={false}
              rounded={false}
              roundedCircle={false}
              src="static/img/map-icon.png"
              thumbnail={false}
            />
            <CardSubtitle
              className="paddlin_left_0"
            >
              madhyamgram
            </CardSubtitle>
          </CardTitle>
          <CardTitle
            className="map_icon d-flex"
          >
            <Image
              alt="call"
              fluid={false}
              rounded={false}
              roundedCircle={false}
              src="static/img/call-icon.png"
              thumbnail={false}
            />
            <CardSubtitle
              className="paddlin_left_0"
            >
              1234567890
            </CardSubtitle>
          </CardTitle>
          <CardTitle
            className="map_icon d-flex"
          >
            <Image
              alt="mail"
              fluid={false}
              rounded={false}
              roundedCircle={false}
              src="static/img/mail-icon.png"
              thumbnail={false}
            />
            <CardSubtitle
              className="paddlin_left_0"
            >
              abc@gmail.com
            </CardSubtitle>
          </CardTitle>
        </Col>
        <Col
          lg={3}
          sm={6}
        >
          <CardText
            className="adderss_text fs-2"
          >
            Doctors
          </CardText>
          <CardText
            className="hiphop_text_1"
          >
            There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour,
          </CardText>
        </Col>
        <Col
          lg={3}
          sm={6}
        >
          <CardText
            className="adderss_text fs-2"
          >
            Useful Links
          </CardText>
          <CardText
            className="Useful_text"
          >
            There are many variations of passages of Lorem Ipsum available, but the majority have suffered ,
          </CardText>
        </Col>
        <Col
          lg={3}
          sm={6}
        >
          <CardText
            className="adderss_text fs-2"
          >
            Newsletter
          </CardText>
          <FormControl
            aria-describedby="basic-addon2"
            aria-label="Enter Your Email"
            className="Enter_text bg-transparent"
            placeholder="Enter Your Email"
          />
          <CardText
            className="subscribe_bt"
          >
            <CardLink
              href="/#"
            >
              Subscribe
            </CardLink>
          </CardText>
          <Card
            body={false}
            className="social_icon py-3 bg-transparent"
            style={
              Object {
                "border": "none",
              }
            }
          >
            <Nav
              fill={false}
              justify={false}
            >
              <NavItem>
                <NavLink
                  className="p-1 m-0"
                  disabled={false}
                  href="/#"
                >
                  <Image
                    alt="fb"
                    fluid={false}
                    rounded={false}
                    roundedCircle={false}
                    src="static/img/fb-icon.png"
                    thumbnail={false}
                  />
                </NavLink>
              </NavItem>
              <NavItem>
                <NavLink
                  className="p-1 m-0"
                  disabled={false}
                  href="/#"
                >
                  <Image
                    alt="twitter"
                    fluid={false}
                    rounded={false}
                    roundedCircle={false}
                    src="static/img/twitter-icon.png"
                    thumbnail={false}
                  />
                </NavLink>
              </NavItem>
              <NavItem>
                <NavLink
                  className="p-1 m-0"
                  disabled={false}
                  href="/#"
                >
                  <Image
                    alt="linkedin"
                    fluid={false}
                    rounded={false}
                    roundedCircle={false}
                    src="static/img/linkedin-icon.png"
                    thumbnail={false}
                  />
                </NavLink>
              </NavItem>
              <NavItem>
                <NavLink
                  className="p-1 m-0"
                  disabled={false}
                  href="/#"
                >
                  <Image
                    alt="instagram"
                    fluid={false}
                    rounded={false}
                    roundedCircle={false}
                    src="static/img/instagram-icon.png"
                    thumbnail={false}
                  />
                </NavLink>
              </NavItem>
            </Nav>
          </Card>
        </Col>
      </Row>
    </Container>
  </Container>
  <Container
    className="copyright_section"
    fluid={true}
  >
    <Container
      fluid={false}
    >
      <Card
        body={false}
        className="copyright_text align-self-center bg-transparent flex-row justify-content-center fs-5"
      >
        2022 All Rights Reserved. Design by
        <CardLink
          href="https://html.design"
        >
          Free html Templates
        </CardLink>
      </Card>
    </Container>
  </Container>
</Fragment>
`;

exports[`snapshot testing --Footer 1`] = `
Array [
  <div
    className="footer_section layout_padding container-fluid"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-3 col-sm-6"
        >
          <div
            className="footer_logo bg-transparent card"
            style={
              Object {
                "border": "none",
              }
            }
          >
            <a
              className="card-link"
              href="index.html"
            >
              <img
                alt="footer"
                className=""
                src="static/img/footer-logo.png"
              />
            </a>
          </div>
          <p
            className="adderss_text fs-2 card-text"
          >
            Contact Us
          </p>
          <div
            className="map_icon d-flex card-title h5"
          >
            <img
              alt="map"
              className=""
              src="static/img/map-icon.png"
            />
            <div
              className="paddlin_left_0 card-subtitle h6"
            />
          </div>
          <div
            className="map_icon d-flex card-title h5"
          >
            <img
              alt="call"
              className=""
              src="static/img/call-icon.png"
            />
            <div
              className="paddlin_left_0 card-subtitle h6"
            />
          </div>
          <div
            className="map_icon d-flex card-title h5"
          >
            <img
              alt="mail"
              className=""
              src="static/img/mail-icon.png"
            />
            <div
              className="paddlin_left_0 card-subtitle h6"
            />
          </div>
        </div>
        <div
          className="col-lg-3 col-sm-6"
        >
          <p
            className="adderss_text fs-2 card-text"
          >
            Doctors
          </p>
          <p
            className="hiphop_text_1 card-text"
          >
            There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour,
          </p>
        </div>
        <div
          className="col-lg-3 col-sm-6"
        >
          <p
            className="adderss_text fs-2 card-text"
          >
            Useful Links
          </p>
          <p
            className="Useful_text card-text"
          >
            There are many variations of passages of Lorem Ipsum available, but the majority have suffered ,
          </p>
        </div>
        <div
          className="col-lg-3 col-sm-6"
        >
          <p
            className="adderss_text fs-2 card-text"
          >
            Newsletter
          </p>
          <input
            aria-describedby="basic-addon2"
            aria-label="Enter Your Email"
            className="Enter_text bg-transparent form-control"
            placeholder="Enter Your Email"
          />
          <p
            className="subscribe_bt card-text"
          >
            <a
              className="card-link"
              href="/#"
            >
              Subscribe
            </a>
          </p>
          <div
            className="social_icon py-3 bg-transparent card"
            style={
              Object {
                "border": "none",
              }
            }
          >
            <div
              className="nav"
              onKeyDown={[Function]}
            >
              <div
                className="nav-item"
              >
                <a
                  className="p-1 m-0 nav-link"
                  data-rr-ui-event-key="/#"
                  disabled={false}
                  href="/#"
                  onClick={[Function]}
                >
                  <img
                    alt="fb"
                    className=""
                    src="static/img/fb-icon.png"
                  />
                </a>
              </div>
              <div
                className="nav-item"
              >
                <a
                  className="p-1 m-0 nav-link"
                  data-rr-ui-event-key="/#"
                  disabled={false}
                  href="/#"
                  onClick={[Function]}
                >
                  <img
                    alt="twitter"
                    className=""
                    src="static/img/twitter-icon.png"
                  />
                </a>
              </div>
              <div
                className="nav-item"
              >
                <a
                  className="p-1 m-0 nav-link"
                  data-rr-ui-event-key="/#"
                  disabled={false}
                  href="/#"
                  onClick={[Function]}
                >
                  <img
                    alt="linkedin"
                    className=""
                    src="static/img/linkedin-icon.png"
                  />
                </a>
              </div>
              <div
                className="nav-item"
              >
                <a
                  className="p-1 m-0 nav-link"
                  data-rr-ui-event-key="/#"
                  disabled={false}
                  href="/#"
                  onClick={[Function]}
                >
                  <img
                    alt="instagram"
                    className=""
                    src="static/img/instagram-icon.png"
                  />
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="copyright_section container-fluid"
  >
    <div
      className="container"
    >
      <div
        className="copyright_text align-self-center bg-transparent flex-row justify-content-center fs-5 card"
      >
        2022 All Rights Reserved. Design by
        <a
          className="card-link"
          href="https://html.design"
        >
          Free html Templates
        </a>
      </div>
    </div>
  </div>,
]
`;
